### DISCUSSION TITLE
NiceGUI Wishlist - Our most wanted features, enhancements and bugfixes.

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/nicegui#3891] [Announcements]
We're thrilled to announce the launch of our [**NiceGUI Wishlist**](https://github.com/orgs/zauberzeug/projects/6/)! This project board is designed to capture your most valuable ideas, feature requests, enhancements, and bug fixes. Our aim is to collect and prioritize contributions that will significantly improve the NiceGUI project.

What's on the Wishlist?

- High-impact features that benefit our users
- Quick wins and nice-to-have enhancements
- Critical bug fixes

Your input is crucial in shaping the future of NiceGUI. We're excited to collaborate with you on these improvements and make NiceGUI even more powerful and user-friendly.

Thank you for your ongoing support and contributions to the NiceGUI community!

**Comment by yudelevi** [comment]
It would be awesome to see https://github.com/zauberzeug/nicegui/discussions/1606 "Sync app.storage between multiple instances" on this wishlist

**Reply by falkoschindler** [reply]
You're right. It also has quite a few upvotes. I'll add it to the list. 👍🏻

**Comment by aersam** [comment]
I'd like some more advanced NPM Management, so that JavaScript Code is not distributed within Python Packages where they are not updated on security issues

**Reply by falkoschindler** [reply]
That's a great idea, @aersam! I'm just not quite sure about the installation process and how the developer would do the NPM management. Would you mind creating a new discussion for this feature request so that we can discuss it in more detail? Thanks!

**Reply by evnchn** [reply]
If implemented, #4718 can help with that

**Comment by keenanjohnson** [comment]
Love that! I think NiceGUI is serving as an excellent example of running a productive and community driven open-source project.

**Comment by denis-x** [comment]
Hi everyone. I would be nice to have a basic "listener", which execute a callback by variable value change. I am currently use invisible GUI elements for such purpose, but appreciate to have dedicated class.

**Reply by falkoschindler** [reply]
@denis-x I'd suggest to create a new discussion for this feature request, where we can talk about the specific requirement and API in more detail. Thanks!

**Reply by evnchn** [reply]
Is this just what the bindings provide? https://nicegui.io/documentation/section_binding_properties
